### PR TITLE
[Rooms] Fix overflow gallery to use all available space when you are a consumer

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What
Fix logic in FloatingLocalVideoLayout and SpeakerLayout to adjust OverflowGallery to use all available space when there is no local video . Fix ScrollableHorizonalGallery to use the entire width when there is no local video.

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3513053

# How Tested
Local test app testing:
https://github.com/Azure/communication-ui-library/assets/79475487/553d1791-91d3-4707-bf32-8eef39cabfd9

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/VideoGallery/OverflowGallery.tsx [hunk 1]
```diff
@@ -94,14 +94,13 @@ export const OverflowGallery = (props: {
...
-          props.layout === 'default'
-            ? `${_convertPxToRem(parentWidth) - 1}rem`
-            : `${_convertPxToRem(parentWidth) - SMALL_FLOATING_MODAL_SIZE_REM.width - 1}rem`
+        width: shouldFloatLocalVideo
+          ? `${_convertPxToRem(parentWidth) - SMALL_FLOATING_MODAL_SIZE_REM.width - 1}rem`
+          : `${_convertPxToRem(parentWidth) - 1}rem`
       };
     }
     return undefined;
-  }, [isNarrow, parentWidth, props.layout]);
```
Comment (JamesBurnside) [inline packages/react-components/src/components/VideoGallery/OverflowGallery.tsx:104]: can layout now be removed from the props? (I dont see it used elsewhere in this file)

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-components/src/components/VideoGallery/FloatingLocalVideoLayout.tsx [hunk 2]
```diff
@@ -208,7 +208,7 @@ export const FloatingLocalVideoLayout = (props: FloatingLocalVideoLayoutProps):
         isShort={isShort}
         onFetchTilesToRender={setIndexesToRender}
         isNarrow={isNarrow}
-        shouldFloatLocalVideo={true}
+        shouldFloatLocalVideo={!!localVideoComponent}
```
Comment (JamesBurnside) [inline packages/react-components/src/components/VideoGallery/FloatingLocalVideoLayout.tsx:211]: Is `localVideoComponent` valid when the video is turned off but the local video avatar shows in the floating form?

**Review by dmceachernmsft — APPROVED** [review]

File: packages/react-components/src/components/VideoGallery/FloatingLocalVideoLayout.tsx (hunk 2 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery/FloatingLocalVideoLayout.tsx:211]: Yup. It is valid when the local video stream is turned off.

File: packages/react-components/src/components/VideoGallery/OverflowGallery.tsx (hunk 1 above)
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery/OverflowGallery.tsx:104]: Good catch. Removed.

**Pull request merged**